File: src/FoodSharing.Site/ClientApp/src/domain/messages/message.ts
import { MessageStatus } from "./messageStatus";

export class Message {
    constructor(
        public id: string,
        public chatId: string,
        public content: string,
        public createdUserId: string,
        public status: MessageStatus,
        public createdDateTimeUtc: Date,
    ) { }

    public updateStatusToRead() {
        this.status = MessageStatus.Read;
    }
}

export function mapToMessage(data: any) {
    return new Message(data.id, data.chatId, data.content, data.createdUserId, data.status, new Date(data.createdDateTimeUtc))
}

File: src/FoodSharing.Site/Models/Chats/Message.cs
namespace FoodSharing.Site.Models.Chats;

public class Message
{
    public Guid Id { get; }
    public Guid ChatId { get; set; }
    public String Content { get; }
    public Guid CreatedUserId { get; }
    public MessageStatus Status { get; }
    public DateTime CreatedDateTimeUtc { get; }

    public Message(
        Guid id, Guid chatId, String content, Guid createdUserId,
        MessageStatus status, DateTime createdDateTimeUtc
    )
    {
        Id = id;
        ChatId = chatId;
        Content = content;
        CreatedUserId = createdUserId;
        Status = status;
        CreatedDateTimeUtc = createdDateTimeUtc;
    }
}